### ISSUE TITLE
Stack Frames not Visible from Breakpoint

### ISSUE THREAD

**Issue by mrwesclarke** [JetBrains/resharper-unity#1671]
Hi, I am using Unity 2019.2.20f1, Rider Plugin 2.0.3 as per the package manager and Rider 2020.1.3 Build #RD-201.7846.1. 

My problem is fairly simple, when hitting a breakpoint in my project I can no longer see a complete call stack to the code, except it says "External Code: [X] frames". None of the code I am asking it to inspect is in external DLLs or external to the project. Tried updating to latest Rider plugin version -> re-importing -> latest Rider etc. 

Only thing that I might be doing that is non-standard is using a subtree for my library files. But, I also deleted that and moved the files into the Scripts dir. No dice. 

Am I missing a setting somewhere? Or something else? This previously worked fine. 

![Screen Shot 2020-05-25 at 16 12 08](https://user-images.githubusercontent.com/7057540/82825583-6c2d7e00-9ea3-11ea-9e9b-8f29459e3855.png)

**Comment by citizenmatt** [comment]
If `Start` is a Unity event function, then you won't see any other stack frames as this method is called by native code. However, it might also be due to Rider incorrectly treating code called by Unity as library code. Uncheck the filter icon in the toolbar, and you'll be able to see more of your project's code. This issue is logged here: [RIDER-43846](https://youtrack.jetbrains.com/issue/RIDER-43846).

**Comment by mrwesclarke** [comment]
Hi, it's not Monobehaviour and in fact a plain old object. But the filter did the trick (could have sworn I clicked that previously). Seems like my code is being treated as a library even when I pulled it back into the main project. Does Rider inspect Git configs for subtrees? 

On the other hand, perhaps I just accidentally clicked the button at some point in the recent past. Thanks for your quick response.

**Comment by citizenmatt** [comment]
I'm not sure on the criteria for the debugger treating files as user or library code, tbh. It should be fixed as part of the issue above. Could you share the full call stack, please, so we can see how it was originally called, as that might play a part in how Rider decides this?

**Comment by mrwesclarke** [comment]
Sure, no problem, picture below. A GameController is an actual MonoBehaviour, but the CommandController is a plain object. All commands are also plain objects. So the only point in this execution where a MB is involved is at the GameController level. 

The code itself (for my framework) sits under Assets/Framework, which is a subtree in the project. Maybe this has something to do with it. 

![Screen Shot 2020-05-25 at 17 17 46](https://user-images.githubusercontent.com/7057540/82829761-f4178600-9eab-11ea-82dd-7d3430ffb00e.png)

**Comment by citizenmatt** [comment]
Thanks. Looks like Rider doesn't treat Unity projects as user code, because the generated msbuild projects use a different folder than where Unity builds the assemblies that get run. Rider doesn't know that these assemblies should be the same. We'll get this fixed.

**Comment by citizenmatt** [comment]
Actually, digging in a bit deeper, Rider handles the different folders fine, as long as the project has already been built in the IDE - it looks at the metadata in the assemblies it knows about, and when they're loaded from another folder, it matches them correctly. But if the project hasn't been built yet, then it doesn't have these output assemblies, and everything gets treated as external code.

In other words, as a quick workaround for now, simply build the project inside Rider. Your code should be treated as "user code" after that.
